velox/exec/tests/AddressableNonNullValueListTest.cpp

```diff
@@ -75,6 +99,57 @@ class AddressableNonNullValueListTest : public testing::Test,
     test::assertEqualVectors(uniqueData, copy);
   }
 
+  // Test copy/appendSerialized round-trip for AddressableNonNullValueList.
+  // Steps in the test:
+  // i) Copy entry length, hash and value of each entry to a stream.
+  // ii) Deserialize stream to a new set of entries.
+  // iii) Read deserialized entries back into a vector.
+  // iv) Validate the result vector.
+  void testSerialization(
+      const std::vector<T>& entries,
+      vector_size_t totalSize,
+      const VectorPtr& uniqueData) {
+    size_t offset = 0;
+    auto buffer = AlignedBuffer::allocate<char>(totalSize, pool());
+    auto* rawBuffer = buffer->as<char>();
```

Comment:
This needs to be asMutable, no?

Reply:
That is better. But then it made me wonder how come this worked. I think that might be because I always added some offset to the pointer for memcpy. wdyt ?